### DISCUSSION TITLE
How to hide/disable "Save changes" button when second tab is opened / document is completed  ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#9845] [Help]
On laravel 10 / filamentphp 3 site I use form with several tabs
```
<?php

use Filament\Facades\Filament;
use Filament\Forms\Components\DatePicker;
use Filament\Forms\Components\Select;
use Filament\Forms\Components\Tabs;
use Filament\Forms\Form;
...
public static function form(Form $form): Form
{

    return $form
        ->schema([

            Tabs::make('Item editor')
                ->tabs([
                    Tabs\Tab::make('Main')
                        ->schema([
                            ...
                                ])
                                ->columns(1),
                        ]),
                    Tabs\Tab::make('Item references')
                        ->schema([
                            ...
                                ])


```
I want when user opens 2nd tab to hide "Save changes" button.

Also when document is completed I want to hide/disable "Save changes" button / show instead of this button message "Can not save" ...

Can I do it ?

```
"laravel/framework": "^10.28.0",
"filament/filament": "^3.0-stable",

```
Thanks in advance!

**Comment by leandrocfe** [comment]
What code did you try?

**Reply by PetroGromovo** [reply]
I did not try any code, as I did not find any hints how that can be done. Serached in docs like "disable Save changes" or "disable Submit button" - but did not find anything like that

**Comment by IkkiRama** [comment]
In your resource folder in file Edit....php, you can add // in ...parent::getFormActions(), olso you can add another button, in here i add close button

```
 protected function getFormActions(): array
    {
        return [
            // ...parent::getFormActions(),
            Action::make('close')->action('saveAndClose'),
        ];
    }
```

**Comment by otavio-araujo** [comment]
Did you get the solution? 

I need the same thing...

**Comment by e2d2-dev** [comment]
Persist Tabs
[https://filamentphp.com/docs/3.x/infolists/layout/tabs#persisting-the-current-tab](url)

On the Page View e.g. create

` protected function getCreateFormAction(): Action
    {
        return parent::getCreateFormAction() 
              ->visible(YOUR URL RECOGNITION CODE HERE)
    }`
